### PR TITLE
Handle failed registers, add metrics, improve logging

### PR DESCRIPTION
**Pull request by rafaelschlatter**
## Summary

Connector does not fail anymore if a register is present in the configuration of the connector but not in the modbus master.

### Added
- Metrics for number of messages sent/received, and for number of connected registers

### Changed
- Connector does not fail if register config is present in connector but not in modbus master
- Exception logging now includes ip, port and register information

### CODE REVIEW

File: Source/Bridge/ModbusBridge.cs [hunk 1]
```diff
@@ -63,10 +67,13 @@ public async Task Connect()
                 {
                     var timer = new Stopwatch();
                     timer.Start();
-                    IEnumerable<Task<(Register, byte[])>> readTasks = _registers
-                        .Select(async r => (r, await _master.Read(r)));
+                    IAsyncEnumerable<(Register r, byte[])> readTasks = _registers
+                        .TrySelect(
+                            async r => (r, await _master.Read(r)),
+                            _ => { });
```
Comment (srudaa) [inline Source/Bridge/ModbusBridge.cs:73]: I don't understand this part. Furthermore, this will also not throw if connection is lost

**Review by srudaa — CHANGES_REQUESTED** [review]
Not able to throw if there is no connection or connection is lost. Furthermore, might consider to make the logs easier to understand and not include the stack trace for known exceptions

File: Source/Bridge/ModbusBridge.cs (hunk 1 above)
Comment (thotokraa) [reply Source/Bridge/ModbusBridge.cs:73]: Well spotted! I believe this can be fixed by rethrowing the exception on line 73.

**Review by thotokraa — CHANGES_REQUESTED** [review]
Looks good. Fix the connectivity issue, then I'll approve.

**Pull request merged**